File: src/main/java/leets/weeth/domain/user/domain/entity/enums/StatusPriority.java
package leets.weeth.domain.user.domain.entity.enums;

import leets.weeth.domain.user.application.exception.StatusNotFoundException;
import lombok.Getter;

@Getter
public enum StatusPriority {
	ACTIVE(1),
	WAITING(2),
	LEFT(3),
	BANNED(4);

	private final int priority;

	StatusPriority(int priority) {
		this.priority = priority;
	}

	public static StatusPriority fromStatus(Status status) {
		if (status == null) {
			throw new StatusNotFoundException();
		}
		return StatusPriority.valueOf(status.name());
	}
}

File: src/main/kotlin/com/weeth/domain/user/domain/entity/enums/StatusPriority.kt
package com.weeth.domain.user.domain.entity.enums

import com.weeth.domain.user.application.exception.StatusNotFoundException

enum class StatusPriority(
    val priority: Int,
) {
    ACTIVE(1),
    WAITING(2),
    LEFT(3),
    BANNED(4),
    ;

    companion object {
        @JvmStatic
        fun from(status: Status?): StatusPriority {
            if (status == null) {
                throw StatusNotFoundException()
            }
            return valueOf(status.name)
        }

        @JvmStatic
        fun fromStatus(status: Status?): StatusPriority = from(status)
    }
}
